### PR TITLE
Feat : Axis Time scale (accept Date type)

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium/react-plot/issues/78

### CODE REVIEW

File: src/components/Axis/Axis.tsx [hunk 1]
```diff
@@ -47,6 +48,14 @@ export interface AxisProps {
...
   hiddenTicks?: boolean;
   tickPosition?: TickPosition;
+  /**
+   * LabelsFormat.
+   * the default value:
+   * in linear and log scale
+   * (value)=>String(value)
+   * in time scale
+   * Smart d3 tickFormat
+   */
```
Comment (stropitek) [inline src/components/Axis/Axis.tsx:58]:
```suggestion
  /**
   * With time scale the default values is d3's smart tickFormat
   * With other types of scales the default is converting the value to a string
   */
```

File: stories/control/timeaxis.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  { x: new Date(2002, 11, 24, 12, 33), y: 40 },
+  { x: new Date(2002, 11, 24, 13, 33), y: 7 },
+  { x: new Date(2002, 11, 24, 14, 33), y: 15 },
+];
+const dataY = [
+  { x: 3, y: new Date(2002, 11, 24, 10, 33) },
+  { x: 6, y: new Date(2002, 11, 24, 11, 33) },
+  { x: 12, y: new Date(2002, 11, 24, 12, 33) },
+  { x: 14, y: new Date(2002, 11, 24, 13, 33) },
+  { x: 15, y: new Date(2002, 11, 24, 14, 33) },
```
Comment (stropitek) [inline stories/control/timeaxis.stories.tsx:34]: If it's supposed to work with numbers (epochs) as well, can you use numbers on one fothe axis?

File: src/components/BarSeries.tsx [hunk 3]
```diff
@@ -100,7 +100,7 @@ function BarSeriesRender({
       {data.map(({ x, y }) => (
         <line
           style={style}
-          key={`${x}-${y}`}
+          key={`${xScale(x)}-${yScale(y)}`}
```
Comment (stropitek) [inline src/components/BarSeries.tsx:103]: This change does not seem related to this PR. Does it fix any sort of bug?

**Comment by wadjih-bencheikh18** [comment]
Yes sorry there is many types error to fix

File: src/components/BarSeries.tsx (hunk 3 above)
Comment (targos) [reply src/components/BarSeries.tsx:103]: I think it's wrong to use x and y as the key. We can replace them with the array index as there is nothing that clearly identifies each data point.

Comment (wadjih-bencheikh18) [reply src/components/BarSeries.tsx]: i used index

File: stories/control/timeaxis.stories.tsx (hunk 2 above)
Comment (wadjih-bencheikh18) [reply stories/control/timeaxis.stories.tsx:34]: yes you can use numbers but it must be in milliseconds
check nasdaq example

**Comment by wadjih-bencheikh18** [comment]
types updates

- **coordinates** : number | string | Date
- **max/min in axis** : number (in time scale we use milliseconds)

**Comment by stropitek** [comment]
I think the support of dates complicate the types and the code a lot. We may want to add support to it but maybe at a later point.

It's easy for the user to iterate over the data and convert it to a number before passing it to react-plot.

Can you revert to accepting numbers only?

**Comment by wadjih-bencheikh18** [comment]
> Can you revert to accepting numbers only?

i couldn't revert but I created a new PR
